File: app/src/renderer/types/zone.ts
// Zone types for Phase 0 prototype

import type { TerminalInZone } from './terminal'

export interface ZoneMetadata {
  [key: string]: unknown
}

export interface Zone {
  id: string
  name: string
  description: string
  metadata: ZoneMetadata
  tags: string[]
  createdAt: string
  updatedAt: string
  planCount?: number
  taskCount?: number
  memoryCount?: number
  terminalCount?: number
}

export interface MemoryInZone {
  id: string
  type: 'Note' | 'Repository' | 'Memory'
  content: string
  metadata: Record<string, unknown>
  tags: string[]
  createdAt: string
  updatedAt: string
  // UI positioning (stored in metadata on the node)
  ui_x?: number
  ui_y?: number
  ui_width?: number
  ui_height?: number
}

export interface TaskInZone {
  id: string
  content: string
  status: 'pending' | 'in_progress' | 'completed' | 'cancelled' | 'blocked'
  metadata: {
    ui_x?: number
    ui_y?: number
    ui_width?: number
    ui_height?: number
    [key: string]: unknown
  }
  tags: string[]
  createdAt: string
  updatedAt: string
  planId: string  // Which plan this task belongs to
  dependsOn: string[]
  blocks: string[]
}

export interface PlanInZone {
  id: string
  name: string
  description: string
  status: 'draft' | 'active' | 'completed' | 'archived'
  metadata: {
    ui_x?: number
    ui_y?: number
    ui_width?: number
    ui_height?: number
    [key: string]: unknown
  }
  tags: string[]
  createdAt: string
  updatedAt: string
  tasks: TaskInZone[]
}

export interface ZoneWithContents extends Zone {
  plans: PlanInZone[]
  memories: MemoryInZone[]
  terminals: TerminalInZone[]
}

File: internal/models/zone.go
package models

import "time"

// Zone represents a workspace container in the graph database.
// Zones organize Plans, Tasks, and Memories into dedicated work areas.
type Zone struct {
	ID          string            `json:"id"`
	Name        string            `json:"name"`
	Description string            `json:"description,omitempty"`
	Metadata    map[string]string `json:"metadata,omitempty"`
	Tags        []string          `json:"tags,omitempty"`
	CreatedAt   time.Time         `json:"created_at"`
	UpdatedAt   time.Time         `json:"updated_at"`
}

// ZoneSearchResult contains a zone with optional related information
type ZoneSearchResult struct {
	Zone    Zone     `json:"zone"`
	Related []string `json:"related,omitempty"`
}

// ZoneWithCounts contains zone data along with counts of contained items
type ZoneWithCounts struct {
	Zone        Zone `json:"zone"`
	PlanCount   int  `json:"plan_count"`
	TaskCount   int  `json:"task_count"`
	MemoryCount int  `json:"memory_count"`
}

// PlanInZone represents a plan with its tasks within a zone context
type PlanInZone struct {
	Plan  Plan         `json:"plan"`
	Tasks []TaskInPlan `json:"tasks"`
}

// MemoryInZone represents a memory within a zone context with UI positioning
type MemoryInZone struct {
	Memory Memory `json:"memory"`
	// UI positioning is stored in Memory.Metadata as ui_x, ui_y, ui_width, ui_height
}

// ZoneWithContents contains a zone with all its plans, tasks, and memories
type ZoneWithContents struct {
	Zone     Zone           `json:"zone"`
	Plans    []PlanInZone   `json:"plans"`
	Memories []MemoryInZone `json:"memories"`
}
